utilities/src/main/java/io/syndesis/qe/resource/ResourceFactory.java

```diff
@@ -0,0 +1,62 @@
+package io.syndesis.qe.resource;
+
+import static org.assertj.core.api.Assertions.fail;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class ResourceFactory {
+    private static List<Resource> createdResources = new ArrayList<>();
+
+    /**
+     * Gets (or creates and gets) the instance of given class and calls its the deploy method.
+     * @param clazz class to create
+     * @param <T> type
+     */
+    public static <T extends Resource> void create(Class<T> clazz) {
+        get(clazz).deploy();
+    }
+
+    public static <T extends Resource> void destroy(Class<T> clazz) {
+        get(clazz).undeploy();
+    }
+
+    /**
+     * Gets (or creates and gets) the instance of given class and returns it.
+     * @param clazz class to create
+     * @param <T> type
+     * @return instance of given class
+     */
+    public static <T extends Resource> T get(Class<T> clazz) {
+        Optional<Resource> oExtRes = createdResources.stream().filter(clazz::isInstance).findAny();
+        if (oExtRes.isPresent()) {
```

Comment:
What if I want to deploy two instances of the same resource? I don't think we do that currently, but sounds like a plausible use case to me

Reply:
I created https://issues.redhat.com/browse/FUSEQE-7545 for that